### PR TITLE
CATROID-1142 Refactor SoundListFragment and related files to kotlin

### PR DESCRIPTION
**Pull request by khushal1707**
https://jira.catrob.at/browse/CATROID-1142

Refactored files:
- SoundListFragment
- SoundController
- SoundAdapter

Bugs: 
- while one sound is playing, on pressing another sound it pauses the player without changing the image -> now it pause the current playing sound with correct change in image and another song is being played on pressings play. 
- The sound now stops when we press back and leave soundListFragment.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    private val soundController = SoundController()
+    private val projectManager: ProjectManager by inject()
+
+    companion object {
+        @JvmField
+        val TAG = SoundListFragment::class.java.simpleName
+    }
+
+    override fun initializeAdapter() {
+        SnackbarUtil.showHintSnackbar(activity, R.string.hint_sounds)
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt:55]:
```suggestion
        SnackbarUtil.showHintSnackbar(requireActivity(), R.string.hint_sounds)
```

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                BackpackListManager.getInstance().backpackedSounds.add(soundController.pack(item))
+                BackpackListManager.getInstance().saveBackpack()
+                packedItemCnt++
+            } catch (e: IOException) {
+                Log.e(TAG, Log.getStackTraceString(e))
+            }
+        }
+        if (packedItemCnt > 0) {
+            ToastUtil.showSuccess(
+                activity, resources.getQuantityString(
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt:86]:
```suggestion
                requireContext(), resources.getQuantityString(
```

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            switchToBackpack()
+        }
+        finishActionMode()
+    }
+
+    override fun isBackpackEmpty(): Boolean =
+        BackpackListManager.getInstance().backpackedSounds.isEmpty()
+
+    override fun switchToBackpack() {
+        val intent = Intent(activity, BackpackActivity::class.java)
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt:101]:
```suggestion
        val intent = Intent(requireContext(), BackpackActivity::class.java)
```

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+            try {
+                adapter.add(soundController.copy(item, currentScene, currentSprite))
+                copiedItemCnt++
+            } catch (e: IOException) {
+                Log.e(TAG, Log.getStackTraceString(e))
+            }
+        }
+        if (copiedItemCnt > 0) {
+            ToastUtil.showSuccess(
+                activity, resources.getQuantityString(
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt:122]:
```suggestion
                requireContext(), resources.getQuantityString(
```

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        if (actionModeType != NONE) {
+            return
+        }
+
+        if (!BuildConfig.FEATURE_POCKETMUSIC_ENABLED) {
+            return
+        }
+
+        if (item?.file?.name?.matches(".*MUS-.*\\.midi".toRegex()) == true) {
+            val intent = Intent(activity, PocketMusicActivity::class.java)
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt:176]:
```suggestion
            val intent = Intent(requireContext(), PocketMusicActivity::class.java)
```

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+            try {
+                soundController.delete(item)
+            } catch (e: IOException) {
+                Log.e(TAG, Log.getStackTraceString(e))
+            }
+            adapter.remove(item)
+        }
+
+        ToastUtil.showSuccess(
+            activity, resources.getQuantityString(
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/SoundListFragment.kt:148]:
```suggestion
            requireContext(), resources.getQuantityString(
```

**Review by Walcho1125 — CHANGES_REQUESTED** [review]
Well done! pls take a look at the remarks

**Comment by khushal1707** [comment]
Done!

**Review by Walcho1125 — APPROVED** [review]
Looks good to me now, well done!

**Comment by ndrnour** [comment]
Generating the APKs is failing for some reason? please check [that.](https://jenkins.catrob.at/job/Catroid/job/PR-4256/)

**Comment by khushal1707** [comment]
Surprisingly, the error got resolved without any changes!

**Review by ndrnour — CHANGES_REQUESTED** [review]
please watch the attached screen recording and fix the bug related to the displayed image holder.

**Comment by ndrnour** [comment]
https://user-images.githubusercontent.com/18437036/127777860-6595316a-2625-4c5f-9ec5-7dfd4c6bc5f7.mp4

**Comment by khushal1707** [comment]
> please watch the attached screen recording and fix the bug related to the displayed image holder.

Few doubts regarding this issue:

- This bug also present in develop, please verify.
- If it's present in develop, do we need to create a ticket for it?
- Do we play 2 or more sound simultaneously in sound list?

**Comment by ndrnour** [comment]
> > please watch the attached screen recording and fix the bug related to the displayed image holder.
> 
> Few doubts regarding this issue:
> 
> * This bug also present in develop, please verify.
> * If it's present in develop, do we need to create a ticket for it?
> * Do we play 2 or more sound simultaneously in sound list?

no need for a ticket, please fix it and mention that in the commit message.
only one sound can be played. 
if you need help with that, let me know.

**Review by ndrnour — APPROVED** [review]

**Comment by ndrnour** [comment]
nice thanks for the extra bug fix

**Pull request merged**